### PR TITLE
Add Witch docs

### PR DESCRIPTION
**Pull request by cyberglot**
Move text from `metastatedev/witch` to `metastate/juvix/doc`.

### CODE REVIEW

File: doc/Witch/intro-to-witch.md
Comment (thealmarty) [inline doc/Witch/intro-to-witch.md:3]: I think we normally do # as the title? Do we want to do that here as well for consistency?
```suggestion
# Introduction to Witch
```

File: doc/Witch/intro-to-witch.md
Comment (thealmarty) [inline doc/Witch/intro-to-witch.md:18]:
```suggestion
We believe that program verification doesn't have to be like this,
```

File: doc/Witch/intro-to-witch.md
Comment (thealmarty) [inline doc/Witch/intro-to-witch.md:26]:
```suggestion
wizard is or what the exact tasks it is supposed to assist with. Wizards seem
```
seems like an awkward sentence here, hope this helps.

File: doc/Witch/intro-to-witch.md
Comment (thealmarty) [inline doc/Witch/intro-to-witch.md:76]: This didn't show in the rendered version?

File: doc/Witch/intro-to-witch.md
Comment (thealmarty) [inline doc/Witch/intro-to-witch.md:37]:
```suggestion
styles (e.g., Agda-style dependent types, 
```

File: doc/Witch/intro-to-witch.md
Comment (thealmarty) [inline doc/Witch/intro-to-witch.md:63]:
```suggestion
    functional programming languages. This guarantees that our approach is based
```

File: doc/Witch/intro-to-witch.md
Comment (thealmarty) [inline doc/Witch/intro-to-witch.md:99]:
```suggestion
    meta-programming<sup>1</sup> constructs that are handled at top-level<sup>2</sup>.
```
Maybe making these superscripts will look better?

**Review by thealmarty — CHANGES_REQUESTED** [review]
I have a few suggestions. Hope they help!

File: doc/Witch/intro-to-witch.md
Comment (cwgoes) [inline doc/Witch/intro-to-witch.md:11]:
```suggestion
*Our hypothesis is that programmers, are
```

File: doc/Witch/intro-to-witch.md
Comment (cwgoes) [inline doc/Witch/intro-to-witch.md:70]:
```suggestion
semantically equivalent to but a syntactic improvement over `handler(operation, params)`, since effect handling
```

File: doc/Witch/intro-to-witch.md
Comment (cwgoes) [inline doc/Witch/intro-to-witch.md:90]:
```suggestion
    IO; we believe it should suffice for the user to write, e.g. `SMT`, and
```

**Review by cwgoes — CHANGES_REQUESTED** [review]
Just a few minor nits

File: doc/Witch/intro-to-witch.md
Comment (cyberglot) [reply doc/Witch/intro-to-witch.md:3]: Sorry, my env auto-gen markdown, and that's usually the format static generators use.

File: doc/Witch/intro-to-witch.md
Comment (cyberglot) [reply doc/Witch/intro-to-witch.md:74]: Err, forgot to add when moved it away from witch's repo. Thanks for catching it.

File: doc/Witch/intro-to-witch.md
Comment (cyberglot) [reply doc/Witch/intro-to-witch.md:97]: That's how GitHub-flavoured markdown does footnotes, we can change what's within brackets, but there's nothing else we can do.

File: doc/Witch/intro-to-witch.md
Comment (cyberglot) [reply doc/Witch/intro-to-witch.md:9]: I'd prefer to keep "you" rather than "programmers"? I think it should sound like a direct conversation than some sort of impersonal technical piece. Idk, I think it feels better.

Comment (cyberglot) [reply doc/Witch/intro-to-witch.md:9]: @thealmarty what do you think?

Comment (thealmarty) [reply doc/Witch/intro-to-witch.md:9]: I think keeping the "you" is fine if that's what you prefer. It matches the style with the rest of the docs.

**Review by thealmarty — APPROVED** [review]
Nice doc!

**Review by cwgoes — CHANGES_REQUESTED** [review]
After some discussions with the communications team, we want to clarify branding a little bit here just so it's clear that Witch is a part of Juvix, not something we're releasing as a standalone product (in contrast, for example, to the MASP) - Witch is a feature like typeclasses or the module system, this will be clearer to users (the research can still be named separately).

Accordingly, in the docs, we should label this section "Writing proofs in Juvix" or something like that and call it the "Juvix witch" or say "the Juvix proof assistant witch" initially. We can still reference the paper, and the content here is fine, but we should clarify the naming conventions.

@awasunyin Let me know if I missed anything.

**Comment by cyberglot** [comment]
That is reasonable, I will update the docs.

**Review by cwgoes — APPROVED** [review]
ACK, this seems fine to me.

Let's wait for final sign-off from @awasunyin

**Review by awasunyin — CHANGES_REQUESTED** [review]
Nice docs! 

Could we remove the graphic `with.svg` and add a link to the repository https://github.com/metastatedev/witch, where people can see the graphic directly? 

Ideally, we should eventually have a version of this .svg that is consistent with the design Juvix' visual identity and include this one on Juvix' docs.

**Comment by cyberglot** [comment]
@awasunyin sure I can remove it, but then I will need to rewrite some bits of text that references the image.

**Review by cwgoes — APPROVED** [review]
ACK from my end

**Comment by cwgoes** [comment]
Checked out of band with @awasunyin :+1:

**Pull request merged**
